src/xhr.ts

```diff
@@ -41,6 +41,52 @@ export class JQueryRequestor extends Requestor {
   }
 }
 
+
+/**
+ * Uses fetch API to make Ajax requests
+ */
+export class FetchRequestor extends Requestor {
+  xhr<T>(settings: JQueryAjaxSettings): Promise<T> {
+    let url: URL = new URL(<string>settings.url);
+    let requestInit: RequestInit = {};
+    requestInit.method = settings.method;
+    requestInit.mode = 'cors';
```

Comment:
I don't think this is needed unless the request is expected to be anything other than HEAD, GET, or POST, or use non simple headers, which for the purpose of this library, it should never be.

reference: https://developer.mozilla.org/en-US/docs/Web/API/Request/mode

Reply:
According to the documentation, when using `RequestInit` constructor, the default is `no-cors`. Won't that prevent us from accessing the properties of the `Response` unless we explicitly opt-in to `cors`?